### ISSUE TITLE
Second level settext headers are colored inconsistently

### ISSUE THREAD

**Issue by joelostblom** [vim-pandoc/vim-pandoc-syntax#141]
Hi, thanks for making this extension. I am new to pandoc, markdown and vim (yes, quite the novice) and your work is really making things easier for me. I am a little confused regarding the highlighting of second level settext headers in the image below.![attached](https://cloud.githubusercontent.com/assets/4560057/7335388/51c4edec-eb88-11e4-9cec-44ff02eee954.png) Why do I need I need a space between the header and the paragraph only for second level settext headers? Also, the header color of settext headers is updated only when running the cursor over them, while the atx headers are highlighted right away.

Bonus question: Why are settext and atx underlined? Sorry if I am missing something obvious here.

**Comment by fmoralesc** [comment]
Second level settext headers like the one in your sample seem to be highlighted as tables, and the ambiguity goes away with the empty line. 

This should be improved, but detecting tables vs. this sort of header is pretty hard, oddly enough. :/ 

tldr; settext headers are wrong for America. 

On the bonus question: I think those words are underlined as spell errors, but I'm not sure.

**Comment by joelostblom** [comment]
Thanks for the prompt reply!
I had a look at the [pandoc table section](http://pandoc.org/README.html#tables) and it seems like when the dashes are used to create a table, there is almost always another character in that line as well (space, plus sign, pipe, colon). Can't the check be whether the line consists of solely dashes or mixed with other characters?

This would only cause problems in one of the eight (the third) table examples demonstrated in the pandoc docs. And I imagine it is more common to use settext headers than creating tables and there are also several alternatives for tables, so maybe that detection criteria would be useful and the header highlighting could be the default in this case instead of the table highlighting?

An additional check could be to compare if the dashed lines consists of as many characters as the line above, but this might be less useful as I imagine people might not always use the same length for their underline and header (but I like doing so for now =)).

Also thanks for the spellcheck hint, didn't know that was included, sweet!

**Comment by joelostblom** [comment]
Just for the record, I have started using ATX headers now and find them easier to use than settext headers in general. So although the syntax highlighting is not fixed, I will go ahead and close this issue since I am the only one who has expressed interest in it. If someone thinks this is an important addition, please feel free to reopen this issue.
